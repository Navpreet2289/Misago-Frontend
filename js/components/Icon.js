// @flow
import classNames from 'classnames'
import * as React from 'react'

export type Props = {
  className?: string,
  name: string
}

const Icon = ({ className, name }: Props) => {
  return (
    <span className={classNames('material-icons', className)}>
      {name.replace(' ', '_')}
    </span>
  )
}

export default Icon