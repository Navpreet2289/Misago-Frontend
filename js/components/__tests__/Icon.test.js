import React from 'react'
import renderer from 'react-test-renderer'
import Icon from '../Icon'

test('<Icon /> component renders', () => {
  const component = renderer.create(
    <Icon name="check" />
  )

  expect(component).toMatchSnapshot()
})

test('<Icon /> component renders with extra className', () => {
  const component = renderer.create(
    <Icon className="lorem ipsum" name="check" />
  )

  expect(component).toMatchSnapshot()
})

test('<Icon /> component normalizes icon names', () => {
  const component = renderer.create(
    <Icon name="check box" />
  )

  expect(component).toMatchSnapshot()
})