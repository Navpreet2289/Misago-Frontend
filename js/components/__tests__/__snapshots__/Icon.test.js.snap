// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> component normalizes icon names 1`] = `
<span
  className="material-icons"
>
  check_box
</span>
`;

exports[`<Icon /> component renders 1`] = `
<span
  className="material-icons"
>
  check
</span>
`;

exports[`<Icon /> component renders with extra className 1`] = `
<span
  className="material-icons lorem ipsum"
>
  check
</span>
`;
